rusk-wallet/src/clients.rs

```diff
@@ -53,188 +41,87 @@ const STAKE_CONTRACT: &str =
-        let res = self.prover.call(2, "rusk", &prove_req).wait()?;
-
-        self.status("Stct proof success!");
-
-        let mut proof_bytes = [0u8; Proof::SIZE];
-        proof_bytes.copy_from_slice(&res);
-
-        let proof = Proof::from_bytes(&proof_bytes)?;
-        Ok(proof)
-    }
-
-    /// Request a WFCT proof.
-    fn request_wfct_proof(
-        &self,
-        commitment: JubJubAffine,
-        value: u64,
-        blinder: JubJubScalar,
-    ) -> Result<Proof, Self::Error> {
-        let mut buf = [0; WFCT_INPUT_SIZE];
-        let mut writer = &mut buf[..];
-        writer.write(&commitment.to_bytes())?;
-        writer.write(&value.to_bytes())?;
-        writer.write(&blinder.to_bytes())?;
-
-        self.status("Requesting wfct proof...");
-        let prove_req = RuskRequest::new("prove_wfct", buf.to_vec());
-        let res = self.prover.call(2, "rusk", &prove_req).wait()?;
-        self.status("Wfct proof success!");
-
-        let mut proof_bytes = [0u8; Proof::SIZE];
-        proof_bytes.copy_from_slice(&res);
-
-        let proof = Proof::from_bytes(&proof_bytes)?;
-        Ok(proof)
-    }
-}
+/// A prover struct that has the `Prove` trait from executio-core implemented.
+/// It currently uses a hardcoded prover which delegates the proving to the
+/// `prove_execute`
+pub struct Prover;
 
-impl Prover {
-    fn status(&self, text: &str) {
-        (self.status)(text)
+impl Prove for Prover {
+    fn prove(
+        tx_circuit_vec_bytes: &[u8],
+    ) -> Result<Vec<u8>, ExecutionCoreError> {
+        Ok(tx_circuit_vec_bytes.to_vec())
     }
 }
 
-/// Implementation of the StateClient trait from wallet-core
-/// inner is an option because we don't want to open the db twice and lock it
-/// We construct StateStore twice
-pub struct StateStore {
-    inner: Mutex<InnerState>,
+/// The state struct is responsible for managing the state of the wallet
+pub struct State {
+    cache: Mutex<Arc<Cache>>,
     status: fn(&str),
-    pub(crate) store: LocalStore,
-}
-
-struct InnerState {
     client: RuskHttpClient,
-    cache: Arc<Cache>,
+    prover: RuskHttpClient,
+    store: LocalStore,
+    pub sync_rx: Option<Receiver<String>>,
 }
 
-impl StateStore {
+impl State {
     /// Creates a new state instance. Should only be called once.
     pub(crate) fn new(
-        client: RuskHttpClient,
         data_dir: &Path,
-        store: LocalStore,
         status: fn(&str),
+        client: RuskHttpClient,
+        prover: RuskHttpClient,
+        store: LocalStore,
     ) -> Result<Self, Error> {
-        let cache = Arc::new(Cache::new(data_dir, &store, status)?);
-        let inner = Mutex::new(InnerState { client, cache });
+        let cfs = (0..MAX_ADDRESSES)
+            .flat_map(|i| {
+                let pk: PhoenixPublicKey =
+                    derive_phoenix_pk(store.get_seed(), i as u8);
+
+                [format!("{:?}", pk), format!("spent_{:?}", pk)]
```

Comment:
@moCello this is related to the cache of the wallet

"Cfs" stands for "ColumnFamilieS" and its ok to call it "spent" since it store the notes that already have a nullifier present in the piecrust state

Column families are very useful to add additional indexes and they are related to a "store logic" that shouldn't be touched by this PR (IMO)

However, since the moonlight introduction, I agree that we should rename at least into "phoenix_spent_<address>"

Reply:
Lets change it when we add moonlight support